src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/InvokeDotNetMethodWithJsonMarshalling.ts

```diff
@@ -0,0 +1,183 @@
+import { platform } from '../Environment';
+import { System_String, Pointer } from '../Platform/Platform';
+import { getRegisteredFunction } from './RegisteredFunction';
+import { error } from 'util';
+
+export interface MethodOptions {
+  type: TypeInstance;
+  method: MethodInstance;
+}
+
+// Keep in sync with InvocationResult.cs
+export interface InvocationResult {
+  succeeded: boolean;
+  result?: any;
+  message?: string;
+}
+
+export interface MethodInstance {
+  name: string;
+  typeArguments?: { [key: string]: TypeInstance }
+  parameterTypes?: TypeInstance[];
+}
+
+export interface TypeInstance {
+  assembly: string;
+  name: string;
+  typeArguments?: { [key: string]: TypeInstance };
+}
+
+export function invokeDotNetMethod<T>(methodOptions: MethodOptions, ...args: any[]): (T | null) {
+  return invokeDotNetMethodCore(methodOptions, null, ...args);
+}
+
+let registrations = {};
+
+function resolveRegistration(methodOptions: MethodOptions) {
+  let existing = registrations[methodOptions.type.assembly];
+  existing = existing && registrations[methodOptions.type.name];
+  existing = existing && registrations[methodOptions.method.name];
```

Comment:
From looking at this, it seems like `registrations` is a dictionary that contains all of assembly, type and method names as keys. Is that that case? I expected some kind of nested lookup here, because the strings could conflict right?

Apologies if you've already discussed this, I haven't been following along for the whole PR.

Reply:
Or maybe I'm misunderstanding how `&&` works....

Reply:
This is an actual bug, it should be existing[...] all over.
I probably don't hit this code path so that's why I don't see it triggered in my tests.